src/smt.rs

```diff
@@ -1018,12 +1018,9 @@ impl SparseMerkleTree {
                     &info.key_bins[idx][0],
                     &info.value_bins[idx][0],
                 ));
-                return Ok((vec![new_leaf], vec![info.structure_pos.into()]));
+                return Ok(Some((new_leaf, info.structure_pos.into())));
```

Comment:
if the return type turns to `StructurePosition` so we can remove the `into` function

Reply:
Done.